care/facility/migrations/0415_auto_20240227_0130.py

```diff
@@ -0,0 +1,38 @@
+# Generated by Django 4.2.10 on 2024-02-26 20:00
+
+from django.db import migrations
+from django.db.models import Count, F, Value, Window
+from django.db.models.functions import RowNumber
+
+
+def fix_duplicate_patient_numbers(apps, schema_editor):
+    PatientConsultation = apps.get_model("facility", "PatientConsultation")
+
+    window = Window(
+        expression=RowNumber(),
+        partition_by=[F("patient_no"), F("facility")],
+        order_by=F("id").asc(),
+    )
+
+    consultations = PatientConsultation.objects.annotate(row_number=window).filter(
+        row_number__gt=1
+    )
+
+    for consultation in consultations:
+        consultation.patient_no = (
+            f"{consultation.patient_no}_{consultation.row_number - 1}"
+        )
```

Comment:
@aparnacoronasafe Existing duplicate patient numbers within a facility will be suffixed with a sequentially incrementing number.